### DISCUSSION TITLE
Unclear about how to work with the VK_KHR_portability_subset extension

### DISCUSSION THREAD

**Discussion by YigitCanAkcay** [KhronosGroup/Vulkan-Portability#17] [Q&A] [answered]
Hello everybody!

First of all I am just learning about Vulkan so maybe that is where my confusion stems from. I am unsure at which exact point to query for the `VK_KHR_portability_subset` extension.

`VK_KHR_portability_subset` device extension requires the `VK_KHR_get_physical_device_properties2` instance extension. To query my physical device and find out whether I must activate `VK_KHR_portability_subset` I need to create an instance first, right? With that in mind the only way I see right now is to activate `VK_KHR_get_physical_device_properties2` in any case and then query for (and activate if possible) `VK_KHR_portability_subset`.

I might be missing something but this doesn't seem optimal. Is there a better way?

**Comment by KarenGhavam-lunarG** [comment]
vkcube (cube.cpp in the Vulkan SDK or found in the KhronosGroup/Vulkan-Tools repository) queries for the portability subset. You could use it as an example.

**Reply by YigitCanAkcay** [reply]
Yes I have looked at that example already and I was also able to figure out how to do this correctly. 

I was just confused that I must activate `VK_KHR_get_physical_device_properties2` to activate `VK_KHR_portability_subset`, but the former is at instance level and the latter at device level, i.e. before being able to know whether I must activate the latter, I must have already activated the former.

@sfricke-samsung s link cleared up my problem with this, though. Activating `VK_KHR_get_physical_device_properties2`, while not mandatory, is the right thing to do going forward.

**Comment by sfricke-samsung** [comment, marked as answer]
There is a write up in the Vulkan Guide for querying and enabling extensions.

But you will first need to query and enabled `VK_KHR_get_physical_device_properties2` at Instance creation time and then `VK_KHR_portability_subset`  at device creation time

https://github.com/KhronosGroup/Vulkan-Guide/blob/master/chapters/enabling_extensions.md
